src/Microsoft.AspNet.Mvc.Core/Controllers/ServiceBasedControllerActivator.cs

```diff
@@ -13,19 +13,26 @@ namespace Microsoft.AspNet.Mvc.Controllers
     public class ServiceBasedControllerActivator : IControllerActivator
     {
         /// <inheritdoc />
-        public object Create(ActionContext actionContext, Type controllerType)
+        public object Create(ControllerContext actionContext)
         {
             if (actionContext == null)
             {
                 throw new ArgumentNullException(nameof(actionContext));
             }
 
-            if (controllerType == null)
-            {
-                throw new ArgumentNullException(nameof(controllerType));
-            }
+            var controllerType = actionContext.ActionDescriptor.ControllerTypeInfo?.AsType();
 
             return actionContext.HttpContext.RequestServices.GetRequiredService(controllerType);
         }
+
+        /// <inheritdoc />
+        public virtual void Release(ControllerContext context, object controller)
+        {
+            var disposable = controller as IDisposable;
+            if (disposable != null)
+            {
+                disposable.Dispose();
+            }
```

Comment:
This method should be a no-op. The DI is responsible for doing disposal if necessary.